sunpy/coordinates/frames.py

```diff
@@ -169,17 +173,26 @@ class BaseHeliographic(SunPyBaseCoordinateFrame):
                                 RepresentationMapping('d_distance', 'd_radius', u.km/u.s)],
     }
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    rsun = QuantityAttribute(default=_RSUN, unit=u.km)
 
-        self._make_3d()
+    def make_3d(self):
+        """
+        This method calculates the third coordinate of the frame.
+        It assumes that the coordinate point is on the surface of the Sun.
 
-    def _make_3d(self):
-        # Make 3D if specified as 2D
+        Returns
+        -------
+        new_frame : `~sunpy.coordinates.frames.BaseHeliographic`
+            A new frame instance with all the attributes of the original but
```

Comment:
It should be clarified that if the coordinate is already 3D, it is simply returned and is not a new instance.

Reply:
Now explained earlier